metricflow/query/query_parser.py

```diff
@@ -217,7 +217,10 @@ def _parse_and_validate_query(
         else:
             parsed_where_constraint = where_constraint
 
-        metric_specs = self._parse_metric_names(metric_names)
+        metric_references = self._parse_metric_names(metric_names)
+        metric_specs = tuple(
```

Comment:
Maybe we make the metric spec from the reference inline here instead of doing it in MetricSemantics? Then we can remove that API.

Might be good to have MetricSpec have a `.from_reference` factory method if it doesn't already.